draft-aegis-mimi-arch-latest.md

```diff
@@ -129,27 +134,28 @@ organization). Besides this generic configuration a provider can also have
 additional custom opaque parameters.
 
 ~~~
-uint16 IdentityProviderType;
-
-struct {
-    IdentityProviderType type;
-    bool supports_accounts;
-    bool supports_domains;
-    opaque parameters<V>;
-} IdentityConfiguration;
+IdentityConfiguration
+{
+    "identity_provider_type": uint8,
+    "supports_accounts": boolean,
+    "supports_domains": boolean,
+    "parameters": object<V> 
+}
 ~~~
 
 An example interface of an identity provider is as follows:
 
 ~~~
-struct {
-    HPKEPublicKey public_key;
-    Credential credential;
-} Identity;
+Identity
+{
+    "public_key": SignaturePublicKey,
+    "credential": Credential
+}
 
-struct {
-    opaque context<V>
-} ApplicationContext;
+ApplicationContext
+{
+    "context": object<V>
```

Comment:
For this one, let's remove this definition and instead replace where it is used with a generic `Object`. One could say there is a particular interface a generic object needs to conform to, but since we have no requirements anything technically works so no need to give it a name.

Reply:
yup, ok. done.